codeqlAlerts.js

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env node
+"use strict";
+
+require("dotenv").config();
+
+const _ = require("lodash");
+const moment = require("moment");
+const owner = "Sparkpost";
+const fileName = "alerts_" + Date.now() + ".json";
+const Promise = require("bluebird");
+const { Octokit } = require("@octokit/rest");
+const octokit = new Octokit({
+  auth: process.env.GITHUB_TOKEN,
+  // Set GitHub Auth Token in environment variable
+});
+
+const [, , ...args] = process.argv;
+
+//  .number, .created_at, .url, .html_url, .state, .dismissed_by.login, .dismissed_at, .dismissed_reason, .rule.id, .rule.severity, .rule.description, .tool.name, .most_recent_instance.classifications[]]
+async function getCodeAlerts(repos) {
+  return await Promise.map(repos, async ({ name, org }) => {
+    const sortedAlerts = {};
+    const summary = {};
+
+    return await octokit
+      .paginate(octokit.codeScanning.listAlertsForRepo, {
+        owner: org,
+        repo: name,
+      })
+      .then(async (alerts) => {
+        const filteredAlerts = await filterCodeAlerts(alerts);
+
+        for (var i = 0; i < filteredAlerts.length; i++) {
+          var rule = filteredAlerts[i].rule.description;
+          if (!sortedAlerts[rule]) {
+            sortedAlerts[rule] = { count: 1 };
+            sortedAlerts[rule]["createdAt"] = alerts[i].created_at;
+            sortedAlerts[rule]["severity"] = alerts[i].rule.severity;
+          } else {
+            sortedAlerts[rule]["count"] += 1;
+          }
+          summary["errorsCount"] =
+            filteredAlerts[i].rule.severity === "error"
+              ? summary["errorsCount"]
+              : (summary["errorsCount"] || 0) + 1;
+          summary["warningsCount"] =
+            filteredAlerts[i].rule.severity === "warning"
+              ? summary["warningsCount"]
+              : (summary["warningsCount"] || 0) + 1;
```

Comment:
looks like logic is reversed here...? if severity is 'warning', shouldn't it increment the warningsCount?
this is more readable and probably gives you what you're looking for:
```
if (severity === "error") { summary.errorsCount = (summary.errorsCount || 0) + 1 }
if (severity === "warning") { summary.warningsCount = (summary.warningsCount || 0) + 1 }
```